### PR TITLE
Fixing ESLint errors in docs. (2/n, 47% done)

### PR DESCRIPTION
**Pull request by anthonyshew**
## Description
2/n of my stacked diffs to address ESLint errors in docs. PRs to ship before this one:
- https://github.com/vercel/turbo/pull/6812

Notably, this PR does contain more defensive error throwing. Thank you ESLint for making our code more defensive.

### CODE REVIEW

File: docs/pages/api/og.tsx [hunk 1]
```diff
@@ -69,16 +69,22 @@ export default async function openGraphImage(
 ): Promise<ImageResponse> {
   try {
     const [fonts, bg] = await loadAssets();
-    const { searchParams } = new URL(req.url!);
+    // @ts-expect-error -- Shouldn't a request always have a url?
+    const { searchParams } = new URL(req.url);
```
Comment (mehulkar) [inline docs/pages/api/og.tsx:73]: Looks like `NextApiRequest` extends `IncomingRequest` and the type comes from here: 

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/e2988a134ec9fa51deb43487231ac607265ba088/node/index.d.ts#L708-L711

File: docs/pages/api/og.tsx [hunk 2]
```diff
@@ -69,16 +69,22 @@ export default async function openGraphImage(
...
 
     // Start with the default title for the type
     let title = TITLE_FOR_TYPE[type];
 
-    // If there'sa a ?title=<title> query param, always prefer that.
+    // If there's a ?title=<title> query param, always prefer that.
     if (searchParams.has("title")) {
-      title = searchParams.get("title")!.slice(0, 100);
+      // @ts-expect-error -- We just checked .has so we know its there.
+      title = searchParams.get("title").slice(0, 100);
```
Comment (mehulkar) [inline docs/pages/api/og.tsx:87]: strange. TS limitation or is `title` typed as a string?

File: docs/theme.config.tsx [hunk 3]
```diff
@@ -80,19 +89,22 @@ const config: DocsThemeConfig = {
...
-    let ogUrl;
+    let ogUrl: string;
 
     if (asPath === "/") {
       ogUrl = `${SITE_ROOT}/api/og`;
-    } else if (frontMatter?.ogImage) {
+    } else if (frontMatter.ogImage) {
       ogUrl = `${SITE_ROOT}${frontMatter.ogImage}`;
     } else {
+      // eslint-disable-next-line no-nested-ternary -- This is fine.
```
Comment (mehulkar) [inline docs/theme.config.tsx:107]: 👎🏾

**Review by mehulkar — APPROVED** [review]

File: docs/theme.config.tsx (hunk 3 above)
Comment (anthonyshew) [reply docs/theme.config.tsx:107]: Good catch, forgot I left this in. It was this connotation:

image

File: docs/pages/api/og.tsx (hunk 2 above)
Comment (anthonyshew) [reply docs/pages/api/og.tsx:87]: TS limitation, yeah. `searchParams.get` ends up typed as `string | null` so the `.slice` call gets upset.

File: docs/theme.config.tsx (hunk 3 above)
Comment (mehulkar) [reply docs/theme.config.tsx:107]: haha i didn't catch that. i am very bad at reading ternaries

**Pull request merged**